fixed logic so no duplicate users/passwords can be displayed and user…

jessicagentz opened the pull request:
I fixed the logic so user will get an error message if they try to use a username that is already taken and no new user will be created to the database. I also added logic for password duplication, but I don't know if we want to keep this part. Because we are storing the encrypted versions, even if the same password is used the likelihood of the encrypted version being the same is very small.

EthanJolles commented:
Started review

If you don't mind can you add a successful test run to ensure there are no breaking changes here?

Also, I may have missed it but the Boolean repository methods don't seem to be implemented 

Agree we do not want to check password duplication only username

jessicagentz commented:
The boolean method from AdminUserRepository is used in AdminUserDetailService for the check if username exists. I deleted duplicate password check logic. I haven't used the automated tests yet so I will look into that more tonight and see if I can get it to run. I did some basic tests related to the things I changed manually, but I did not test everything possible.

EthanJolles commented:
![image](https://github.com/user-attachments/assets/283c06be-2789-456e-8bd5-a9770335d3bb)
All automated tests passing
